databroker/src/grpc/kuksa_val_v2/val.rs

```diff
@@ -440,6 +640,337 @@ mod tests {
+                    .collect::<std::collections::HashSet<_>>()
+                {
+                    match (response.get(key), expected_response.get(key)) {
+                        (Some(entry1), Some(entry2)) => {
+                            assert_eq!(entry1.value_state, entry2.value_state);
+                        }
+                        (Some(entry1), None) => {
+                            assert!(f, "Key '{}' is only in response: {:?}", key, entry1)
+                        }
+                        (None, Some(entry2)) => assert!(
+                            f,
+                            "Key '{}' is only in expected_response: {:?}",
+                            key, entry2
+                        ),
+                        (None, None) => unreachable!(),
+                    }
+                }
+            }
+            Err(err) => {
+                assert!(f, "Error {:?}", err)
+            }
+        }
+    }
+
+    #[tokio::test]
+    async fn test_publish_value() {
+        let broker = DataBroker::default();
+        let authorized_access = broker.authorized_access(&permissions::ALLOW_ALL);
+        let f = false;
+
+        let entry_id = authorized_access
+            .add_entry(
+                "test.datapoint1".to_owned(),
+                broker::DataType::Bool,
+                broker::ChangeType::OnChange,
+                broker::EntryType::Sensor,
+                "Test datapoint 1".to_owned(),
+                None,
+                None,
+            )
+            .await
+            .unwrap();
+
+        let request = proto::PublishValueRequest {
+            signal_id: Some(proto::SignalId {
+                signal: Some(proto::signal_id::Signal::Id(entry_id)),
+            }),
+            data_point: {
+                let timestamp = Some(std::time::SystemTime::now().into());
+
+                let value = proto::Value {
+                    typed_value: Some(proto::value::TypedValue::Bool(true)),
+                };
+
+                Some(proto::Datapoint {
+                    timestamp,
+                    value_state: Some(proto::datapoint::ValueState::Value(value)),
+                })
+            },
+        };
+
+        // Manually insert permissions
+        let mut publish_value_request = tonic::Request::new(request);
+        publish_value_request
+            .extensions_mut()
+            .insert(permissions::ALLOW_ALL.clone());
+
+        match broker.publish_value(publish_value_request).await {
+            Ok(response) => {
+                // Handle the successful response
+                let publish_response = response.into_inner();
+
+                // Check if there is an error in the response
+                if let Some(error) = publish_response.error {
+                    assert!(f, "Publish failed with error: {:?}", error);
+                } else {
+                    println!("Publish succeeded.");
+                }
```

Comment:
Here you just want to assert that the error is `None` right? You can not assert both with same unit test.
```suggestion
                assert_eq!(publish_response.error, None);
```